tests/posix/CMakeLists.txt

```diff
@@ -93,6 +93,10 @@ function(build_test name srcs)
 	add_check_whitespace(tests-posix-${name} ${CMAKE_CURRENT_SOURCE_DIR}/${srcs})
 	add_executable(${name} ${srcs})
 	target_link_libraries(${name} pmemfile-posix_shared pmemfile_test test_backtrace)
+	#disable dangling else warning caused by bug in gtest
+	if (CMAKE_COMPILER_IS_GNUCC AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7.0)
+		target_compile_options(${name} PRIVATE "-Wno-dangling-else")
+	endif()
```

Comment:
Please submit it as a separate fix. Also don't check for gcc version - just use add_flag function which tests whether flag is supported or not.